sled-agent/src/storage/dump_setup.rs

```diff
@@ -0,0 +1,153 @@
+use crate::storage_manager::DiskWrapper;
+use camino::Utf8PathBuf;
+use illumos_utils::zpool::ZpoolHealth;
+use omicron_common::disk::DiskIdentity;
+use sled_hardware::DiskVariant;
+use slog::Logger;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::MutexGuard;
+
+#[derive(Default)]
+pub struct DumpSetup {
+    savecore_lock: Arc<std::sync::Mutex<()>>,
+}
+
+impl DumpSetup {
+    pub(crate) async fn poll_dumpdev_setup(
```

Comment:
Nitpick: Maybe `update_dumpev_setup`? Calling this `poll` feels a bit misleading, since it's updating state, not querying for a result.